### DISCUSSION TITLE
Exclude specific folders

### DISCUSSION THREAD

**Discussion by AFumi39** [lavinir/hassio-onedrive-backup#42] [Ideas]
I've a suggestion: similarly on sync folders, would be useful to exclude specific folders.

I give an example. I've MariaDB component in my HA Supervised instance. Backup the component is useless because it can't be restored (if you restore the files you'll get inconsistency errors), so I create an external SQL backup with proper tools.
Now the backup include the ~2GB.sql file that I generate plus the ~2GB MariaDB backup (that's useless). If I can exclude the specific folder of MariaDB I can save lot of space from backup.

Thank you

**Comment by lavinir** [comment]
Not sure I completely follow. The MariaDB is backed up as one of the Addons in the regular backups and you also have your own exported backup which you wouldn't want to include ?
Why would the external file be included ? The File Syncs allow you to target specific paths you want to sync.

**Reply by AFumi39** [reply]
(is just an example)
No, MariaDB component can't be backed up/restored using the HA tools because the files turns out to be inconsistent.
You have 2 ways to do it:
1) stop the service before backup so the files remains "freezed" and are not inconsistents anymore
2) do a sql dump with sql tools/commands
I use the option 2) because the backup can take some time, so I don't have to keep MariaDB shutted down for too long.

So, if I can exclude the dpath of MariaDB component, I can save about 2 GB of db (2GB of space both locally and on OneDrive, and also the upload time of 2GB of data) that's practically useless for me.

That's only a suggestion based on my needing, maybe for someone else it can be useful too

**Comment by lavinir** [comment]
You're talking about the Backup not the File Sync ? And you mean it's currently included with the rest of the Addons as part of the regular backups ?

**Reply by AFumi39** [reply]
Sorry, you're right! I was mistaken!
It's an addon, not a custom component 🤦‍♂️

So you can't simply exclude it by path, I think you have to exclude the addon from the backup...

**Reply by lavinir** [reply]
I will add the ability to exclude add-ons from back up to the backlog

**Reply by lavinir** [reply]
This is now available in version 2.0
